### PR TITLE
Feature/se 43

### LINKED ISSUE #43: Create the front end for the Quiz
image

### PR DESCRIPTION
**Pull request by j0world**
closes SE2-WS23/documentation#43

E742C22F-199E-4780-A099-FC7165F8FFAC

### CODE REVIEW

File: src/pages/Quiz/QuizPage.jsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    const theme = useTheme();
+    const [currentQuestion, setCurrentQuestion] = useState(0);
+    const [progressValue, setProgressValue] = useState(1);
+
+    const handleAnswerButtonClick = (answer) => {
+        if (dummyQuestions[currentQuestion].correct === answer.target.textContent) // checks if the answer is correct
+            console.log("Correct");
+        else
+            console.log("Incorrect");
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: You consider to show that as visual representation to the user, but I think that is @NicolasKmbr decision

File: src/pages/Quiz/QuizPage.jsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            setProgressValue((((currentQuestion +1)/ dummyQuestions.length) * 100));
+            return;
+        }
+
+        // update the progress bar on the last question
+        if (currentQuestion === dummyQuestions.length-1)
+            setProgressValue(100); 
+    }
+
+    const nextButtonClick = () => {
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: Can you name this handleNextButtonClick?

File: src/pages/Quiz/QuizPage.jsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        return;
+    }
+
+    const dummyQuestions = [ // placeholder for the real questions
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: Remove that comment and please move this mockData to mockData.js

File: src/pages/Quiz/QuizPage.jsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                    backgroundColor: theme.palette.primary.light,
+                    flexGrow: 1,
+                    display: "flex",
+                    flexDirection: "column",
+                    justifyContent: "center",
+                    alignItems: "center",
+                    }}
+                >
+                    
+                    <h1 style={{margin: "0% 0 15vh 0"}}>
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: Why did you use 0% and 15vh in the same statement?

File: src/pages/Quiz/QuizPage.jsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                        {dummyQuestions[currentQuestion].question}
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: You can consider to also set here a default

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

File: src/pages/Quiz/QuizPage.jsx (hunk 2 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: done

File: src/pages/Quiz/QuizPage.jsx (hunk 3 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: It has been relocated. 
I also added a random question selection which is now incredibly secure >:)
If that needs to leave again, I would completely understand.

File: src/pages/Quiz/QuizPage.jsx (hunk 4 above)
Comment (j0world) [reply src/pages/Quiz/QuizPage.jsx]: changed now :)

File: src/pages/Quiz/QuizPage.jsx
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx:1]: Why can I answer the last question multiple times?

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

File: src/pages/Quiz/QuizPage.jsx [hunk 6]
```diff
@@ -56,7 +57,7 @@ function QuizPage() {
         return userAnswerArray;
     }
 
-    const dummyQuestions = useMemo(() => randomQuestionsSelection(), []); 
+    const dummyQuestions = useMemo(randomQuestionsSelection, []); 
     const userAnswerArray = useMemo(() => userAnswerArrayLength(), []);
```
Comment (Chrissydrx) [inline src/pages/Quiz/QuizPage.jsx]: Remove this () => like in the line above

**Review by Chrissydrx — CHANGES_REQUESTED** [review]

**Review by Chrissydrx — APPROVED** [review]
image

**Pull request merged**